service/storage_proxy.cc

```diff
@@ -3465,9 +3465,12 @@ class abstract_read_executor : public enable_shared_from_this<abstract_read_exec
             return _proxy->query_mutations_locally(_schema, cmd, _partition_range, timeout, _trace_state);
         } else {
             tracing::trace(_trace_state, "read_mutation_data: sending a message to /{}", ep);
-            return _proxy->_messaging.send_read_mutation_data(netw::messaging_service::msg_addr{ep, 0}, timeout, *cmd, _partition_range).then([this, ep](rpc::tuple<reconcilable_result, rpc::optional<cache_temperature>> result_and_hit_rate) {
+            clock_type::time_point sent_at = clock_type::now();
+            _proxy->mark_sent(ep, sent_at);
```

Comment:
If you plan to only remember when you sent *read* requests and recieved responses for *read* requests, then you should call it mark_sent_read() or something. But I wonder - if we're looking for down nodes, not "slow" nodes (and the commit message could use some clarification on this), why only look at read messages? Any received message is an indication that the node is not down. Maybe we can do this mark_sent/mark_responded in some lower level function, and not in multiple read/write message implementations?

Reply:
Yeah, it may make sense. On the other hand, now the extra overhead is reduced to the only interested party - the read mechanism. I'll look for a way to push it to a lower layer though.